ngraph/core/reference/include/ngraph/runtime/reference/convolution_backprop_data.hpp

```diff
@@ -23,217 +19,224 @@ namespace ngraph
-                    // coordinate:
-                    //
-                    //   F
-                    //
-                    // over the range (noninclusive on the right):
-                    //
-                    //   (chan_out,0,0,...,0) ->
-                    //     (chan_out+1,
-                    //      chans_in_count,
-                    //      filter_dims_1,
-                    //        ...,
-                    //      filter_dims_n)
-                    //
-                    // with unit stride.
-
-                    Shape filter_transform_start(2 + n_spatial_dimensions);
-                    Shape filter_transform_end(2 + n_spatial_dimensions);
-
-                    filter_transform_start[filter_out_channel_axis] = out_channel;
-                    filter_transform_end[filter_out_channel_axis] = out_channel + 1;
-                    filter_transform_start[filter_in_channel_axis] = 0;
-                    filter_transform_end[filter_in_channel_axis] = 1;
-
-                    for (size_t i = 2; i < n_spatial_dimensions + 2; i++)
+                    else
                     {
-                        filter_transform_start[i] = 0;
-                        filter_transform_end[i] = filter_shape[i];
+                        for (size_t batch = 0; batch < input_shape[0]; ++batch)
+                        {
+                            for (size_t channel = 0; channel < input_shape[1]; ++channel)
+                            {
+                                for (int i_z = 0; i_z < input_3d[0]; ++i_z)
+                                {
+                                    for (int i_y = 0; i_y < input_3d[1]; ++i_y)
+                                    {
+                                        for (int i_x = 0; i_x < input_3d[2]; ++i_x)
+                                        {
+                                            auto offset = batch * input_size * input_shape[1] +
+                                                          channel * input_size;
+
+                                            input_zeros.push_back(
+                                                in[offset + i_x + i_y * input_3d[2] +
+                                                   i_z * input_3d[2] * input_3d[1]]);
+
+                                            if (i_x < input_3d[2] - 1)
+                                            {
+                                                for (int k = 0; k < strides_3d[2] - 1; k++)
+                                                {
+                                                    input_zeros.push_back(0);
+                                                }
+                                            }
+                                        }
+
+                                        if (i_y < input_3d[1] - 1)
+                                        {
+                                            for (int y_dim = 0;
```

Comment:
Consider `vector::resize`

Reply:
`for` loop is replaced with insert function.